src/App.tsx

```diff
@@ -1,24 +1,74 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { FilterBy } from './enums/FilterBy';
+import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api/todos';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+
+const USER_ID = 10321;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const activeTodosNumber = todos
+    .filter(todo => todo.completed === false).length;
+  const completedTodosNumber = todos.length - activeTodosNumber;
+
+  const getFilteredTodos = (filter: FilterBy) => {
```

Comment:
just make it like `useMemo`
```suggestion
  const preparedTodos = useMemo(() => {
```